contacts.py

```diff
@@ -555,54 +555,57 @@ def send_status(self, data, args):
         # These are fields allowed in the send_status, and just copied from top level into each data point
         # Note memo is not supported yet and is a placeholder
         repeated_fields = {k: data.get(k) for k in ['memo', 'replaces', 'status'] if data.get(k)}
-        return self.send_or_sync(data, repeated_fields)
+        self.send_or_sync(data, repeated_fields)
+        return {"status": "ok"}
 
     # Common part of both /status/send and sync reception
-    def send_or_sync(self, data, repeated_fields):
-        floating_seconds = current_time()
+    def send_or_sync(self, data, repeated_fields=None, floating_seconds=None):
+        if not floating_seconds:
+            floating_seconds= current_time()
         serial_number = 0
         # first process contacts, then process geocode
         for contact in data.get('contact_ids', []):
-            contact.update(repeated_fields)
+            contact.update(repeated_fields or {})
             self._insert_blob_with_optional_replacement(self.contact_dict, contact, floating_seconds, serial_number)
             # increase by two each time to deal with potential second insert
             serial_number += 2
         for location in data.get('locations', []):
-            location.update(repeated_fields)
+            location.update(repeated_fields or {})
             self._insert_blob_with_optional_replacement(self.spatial_dict, location, floating_seconds, serial_number)
             # increase by two each time to deal with potential second insert
             serial_number += 2
-        return {"status": "ok"}
+        return
 
     def _update(self, update_token, updates, floating_time, serial_number):
         for this_dict in self.contact_dict, self.spatial_dict:
             this_dict.update(update_token, updates, floating_time, serial_number)
 
     # status_update POST
-    # { locations: [{min_lat,update_token,...}], contacts:[{id,update_token, ... }], memo, replaces, status, ... ]
+    # { locations: [{min_lat,update_token,...}], contacts:[{id,update_token, ... }], update_tokens: [ut,...], replaces, status, ... ]
     @register_method(route='/status/update')
     def status_update(self, data, args):
         logger.info('in status_update')
-        # floating_seconds = current_time()
-        length = data.get('length')  # This is how many to replace
-        floating_seconds = current_time()
-        serial_number = 0
+        self._update_or_result(serial_number=0, floating_seconds=current_time(), **data)
```

Comment:
you could also default serial_number in _update_or_result to 0 and then just not pass.  Whatever you think makes most symantic sense.

You could also not pass floating_seconds and allow the default.

Reply:
Makes sense as it is because in the other call to self._update_or_result I use the same floating_seconds on two calls.